server/src/graphql-types/ChapterUser.ts

```diff
@@ -25,3 +26,24 @@ export class ChapterUser {
   @Field(() => Boolean, { nullable: true })
   is_bannable?: boolean;
 }
+
+@ObjectType()
+export class UserChapters {
+  @Field(() => Int)
+  chapter_id: number;
+
+  @Field(() => Date)
+  joined_date: Date;
+
+  @Field(() => Boolean)
+  subscribed: boolean;
+
+  @Field(() => ChapterRole)
+  chapter_role: ChapterRole;
+
+  @Field(() => Chapter)
+  chapter: Chapter;
+
+  @Field(() => Boolean, { nullable: true })
+  is_bannable?: boolean;
+}
```

Comment:
Similarly to `UserEvents`, we don't need a new class.

Reply:
Okay, looking at the hoops you've had to jump through to avoid creating a new class, I take it all back.  Sorry, @Sboonny!

I microscopic change like this (expanding the data type fetched by a single query resolver) should not force you to change other query resolvers.  If they needed the extra data, that would be fine, but they do not.

Here's what I think: we go with

```ts
@ObjectType()
class UserEvent extends EventUserWithRelations {
  @Field(() => Event)
  event: Event;
}
```

and the same for `UserChapter` and use those to make 

```ts
@ObjectType()
export class UserInformation extends UserWithPermissions {
  @Field(() => String)
  email: string;

  @Field(() => [UserBanWithRelations])
  user_bans: UserBanWithRelations[];

  @Field(() => [UserChapter])
  user_chapters: UserChapter[];

  @Field(() => [UserEvent])
  user_events: UserEvent[];
}
```

Then we can create a second query `userDownload`, say, that gets all the download data, but only after the user clicks (so we don't waste time getting it for every visit to the profile).  That means there's no need to change anything else about the profile - the button can go in smoothly.

Reply:
what happens when we extends `UserWithPermissions` that has `user_chapters` and `user_events` types to `UserInformation` that has `user_chapters`, and `user_events` types in it? 

Is the first type declaration ignored?

Reply:
It should overwrite them.  The easiest way to find out is just to try it and see, but the docs are [here](https://typegraphql.com/docs/inheritance.html) if you're interested.

Reply:
Ya, I have tested it. just find the idea of overwritten type that you already decided weird, adding and extracting declarations is understandable, but full overwriting it is weirding me out